rustls/src/crypto/mod.rs

```diff
@@ -69,7 +69,7 @@ pub use crate::msgs::handshake::KeyExchangeAlgorithm;
 ///   crate.
 /// - [`crypto::aws_lc_rs::default_provider`]: (behind the `aws_lc_rs` feature,
 ///   which is optional).  This provider uses the [aws-lc-rs](https://github.com/aws/aws-lc-rs)
-///   crate.
+///   crate.  The `fips` crate feature makes this option use FIPS140-3-approved cryptography.
```

Comment:
Does this mean that the build will fail if the version of the aws-lc library being built against isn't FIPS validated for the target platform?

What happens on a target for which the build succeeds, and the version of aws-lc is FIPS-validated for some targets, but not the current target?

And, what happens when the version of aws-lc used in the build isn't FIPS-validated?

What I would expect to happen in all these cases is that the build would fail, since the user is trying to ensure that they get a fips-validated configuration.